app/src/main/java/at_challenge/deaguiar/daniel/atchalleng/controller/RoutesListActivity.java

```diff
@@ -0,0 +1,229 @@
+            i.putExtra(RouteDetailActivity.EXTRA_ROUTE_NAME, route.getLongName());
+
+            startActivity(i);
+        }
+        else {
+            Toast.makeText(getApplicationContext(),
+                    R.string.device_diconnected,
+                    Toast.LENGTH_LONG).show();
+        }
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu_routes_list, menu);
+
+        // Associate searchable configuration with the SearchView
+        SearchManager searchManager = (SearchManager) getSystemService(Context.SEARCH_SERVICE);
+        SearchView searchView = (SearchView) menu.findItem(R.id.search).getActionView();
+        searchView.setSearchableInfo(searchManager.getSearchableInfo(getComponentName()));
+
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        int id = item.getItemId();
+
+        //noinspection SimplifiableIfStatement
+        if (id == R.id.search) {
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    private class RouteAdapter extends ArrayAdapter<Route> {
+
+        public RouteAdapter(ArrayList<Route> routes) {
+            super(getApplicationContext(), 0, routes);
+        }
+
+        @Override
+        public View getView(int position, View convertView, ViewGroup parent) {
+
+            if (convertView == null) {
+                convertView = getLayoutInflater().inflate(R.layout.route_item_list, parent, false);
+            }
+
+            Route route = getItem(position);
+
+            TextView shortNameTextView = (TextView)convertView.findViewById(R.id.list_item_short_name);
+            shortNameTextView.setText(route.getShortName());
+
+            TextView longNameTextView = (TextView)convertView.findViewById(R.id.list_item_long_name);
+            longNameTextView.setText(route.getLongName());
+
+            return convertView;
+        }
+    }
+
+    /**
+     * Check if wifi or mobile connection are available
+     * @return
+     */
+    private boolean internetIsAvailable() {
+        ConnectivityManager manager = (ConnectivityManager)
+                getSystemService(Context.CONNECTIVITY_SERVICE);
+
+        NetworkInfo mobile = manager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE);
+        NetworkInfo wifi = manager.getNetworkInfo(ConnectivityManager.TYPE_WIFI);
+
+        if (wifi != null && wifi.isConnected())
+            return true;
+        if (mobile != null && mobile.isConnected())
+            return true;
+        return false;
+    }
+
+    private class FetchRouteData extends HttpRequesAsyncTask {
```

Comment:
I needed to download the data from a thread different than the main thread
and both AsyncTask and IntentService fitted my needs. Checking the code
again and considering the amount of data that is fetched I think that using
only AsyncTask would be better. I thought in the beginning of the project
in implement using AsyncTask, that's why I've created the abstract class
HttpRequestAsyncTask, in order to create a different subclass for each API
(routes, bus, departure). I had some questions about the best way to
proceed to solve this problem. I just needed to chat a few seconds with
another developer in order to choose the best way.

2015-06-17 13:16 GMT-03:00 André Luiz Cardoso <notifications@github.com>:

> In
> app/src/main/java/at_challenge/deaguiar/daniel/atchalleng/controller/RoutesListActivity.java
> <https://github.com/daniboy000/at_challenge/pull/1#discussion_r32644490>:
>
> > +     */
> > +    private boolean internetIsAvailable() {
> > +        ConnectivityManager manager = (ConnectivityManager)
> > +                getSystemService(Context.CONNECTIVITY_SERVICE);
> > +
> > +        NetworkInfo mobile = manager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE);
> > +        NetworkInfo wifi = manager.getNetworkInfo(ConnectivityManager.TYPE_WIFI);
> > +
> > +        if (wifi != null && wifi.isConnected())
> > +            return true;
> > +        if (mobile != null && mobile.isConnected())
> > +            return true;
> > +        return false;
> > +    }
> > +
> > +    private class FetchRouteData extends HttpRequesAsyncTask {
>
> Why you decided to use an AsyncTask to do the network operation here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/daniboy000/at_challenge/pull/1/files#r32644490>.
>



-- 
Daniel Besen de Aguiar

"Those who wish to succeed must ask the right preliminary questions"


      Aristotle